File: payroll-api/src/main/java/com/company/payroll/mapper/DepartmentMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Department;

@Repository
public interface DepartmentMapper {
    int deleteByPrimaryKey(Integer deptNo);
    int insert(Department row);
    int insertSelective(Department row);
    List<Department> selectList();
    Department selectByPrimaryKey(Integer deptNo);
    int updateByPrimaryKeySelective(Department row);
    int updateByPrimaryKey(Department row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/DepartmentMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Department
import org.springframework.stereotype.Repository

@Repository
interface DepartmentMapper {
  fun deleteByPrimaryKey(deptNo: Int): Int

  fun insert(row: Department): Int

  fun insertSelective(row: Department): Int

  fun selectByPrimaryKey(deptNo: Int): Department

  fun selectList(): List<Department>

  fun updateByPrimaryKeySelective(row: Department): Int

  fun updateByPrimaryKey(row: Department): Int
}